Bottom: edfb01f34c616ef8430e6e0af41faea0f8bb0609
Top:    ca784175a375aa43302c7c3e4bac0cccbd055a6a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-03 23:13:06 +0100

TMP FIXES


---

diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index 19bce70..f418e1a 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -432,8 +432,12 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   int layer_num = GetLayerNumber (PCB->Data, layer);
   int group = GetLayerGroupNumberByPointer (layer);
   struct poly_info info = {drawn_area, layer};
+  bool is_outline;
 
-  if (layer_num < max_copper_layer)
+  is_outline = strcmp (layer->Name, "outline") == 0 ||
+               strcmp (layer->Name, "route") == 0;
+
+  if (layer_num < max_copper_layer && !is_outline)
     {
       r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
       r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
@@ -445,12 +449,6 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   if (TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
 
-  if (layer_num < max_copper_layer)
-    {
-      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
-      r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
-    }
-
   /* draw all visible lines this layer */
   r_search (layer->line_tree, drawn_area, NULL, line_callback, layer);
   r_search (layer->arc_tree,  drawn_area, NULL, arc_callback,  layer);
@@ -460,8 +458,7 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
      auto-outline magically disappear when you first add something to
      the "outline" layer.  */
 
-  if (strcmp (layer->Name, "outline") == 0 ||
-      strcmp (layer->Name, "route") == 0)
+  if (is_outline)
     {
       if (IsLayerEmpty (layer))
         {
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 1f25448..98959fe 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1001,7 +1001,6 @@ ps_set_layer (const char *name, int group, int empty)
      we want to "print outlines", and we have an outline layer,
      print the outline layer on this layer also.  */
   if (global.outline &&
-      global.is_copper &&
       global.outline_layer != NULL &&
       global.outline_layer != PCB->Data->Layer+idx &&
       strcmp (name, "outline") &&
